Refactoring code

mkangoo opened the pull request.

mkangoo commented:
This is not final, I'm working on it.

mkangoo commented on src/core/schedule.js:
```diff
@@ -67,3 +43,26 @@ export const getFullSchedule = shiftWeek => {
 	return scheduleWeek.join('')
 }
+
+const getDayIndex = date => {
```
I made the date parameter to be able to test this function without problems

mkangoo replied:
I made the `date` parameter to be able to test this function without problems

mkangoo reviewed (commented):
It's worth making changes at this point

Ilanaya commented:
> This is not final, I'm working on it.

Ping me, when you'll need review. 
You can also ask me anything in telegram if you need

mkangoo commented on src/core/schedule.js:
```diff
@@ -25,17 +17,7 @@ export const getScheduleOutput = (lessonType, { start, end, title, description,
-
-export function getLessonsForDay(targetDay, scheduleData, shouldShiftWeek = false, weekNumber = getWeekNumber(new Date())) {
+export function getLessonsForDay(targetDay, scheduleData, shouldShiftWeek, weekNumber = getWeekNumber(new Date())) {
 	const lessonToday = scheduleData.find(({ day, events }) => day === targetDay && events)
```
I duplicate this handler in different parts of the code

Ilanaya replied:
Hm, where?)

mkangoo replied:
Sorry, I moved the handler, but did not move the comment, thereby causing confusion. I mean `const statusOffset = JSON.parse(Object.values(weekOffset).join())`

Ilanaya commented on src/inlineKeyboard/inlineKeyboardHandler.js:
```diff
@@ -1,19 +1,19 @@
 		},
 		getScheduleNextWeek: ctx => {
-			ctx.replyWithHTML('<b>Выберите день:</b>', getButtonDaysOfWeek(true))
+			ctx.replyWithHTML('<b>Выберите день:</b>', getButtonDaysOfWeek({ raw: true }))
```
`{ raw: true }` was just an example. You should find appropriate name for a param instead of `raw`
I meant that every function that used to take a boolean param i.e.

```ts
function myFunction(boolOption) {
  return boolOption ? "foo" : "bar"
}
myFunction(true) // <- this is bad as readability is poor
```

```ts
function myFunction(options) {
  const { preferFoo } = options // or directly destructure in param
  return preferFoo ? "foo" : "bar"
}
myFunction({ preferFoo: true }) // <- better readability
```

Ilanaya commented on src/core/schedule.js:
```diff
@@ -67,3 +43,35 @@ export const getFullSchedule = shiftWeek => {
 	return scheduleWeek.join('')
 }
+
+const getDayIndex = () => {
+	return {
+		today() {
+			const date = new Date()
+			const currentHours = date.getUTCHours()
+			// Time zone offset in Moscow
+			date.setUTCHours(currentHours + 3)
+			const currentDay = date.getDay()
+			const dayIndex = currentDay % 7
+			return dayIndex % 7
+		},
+		tomorrow() {
+			return this.today() + 1
+		},
+	}
+}
```
1. This function name doesn't address its return result. I expect it to return an index (number) 
2. Why not add two separate functions `getTodayIndex` and `getTomorrowIndex`? Current version looks too much complicated

mkangoo replied:
1.This function returns the day index
Here is the result of execution
`console.log(getDayIndex().today(new Date()))
`
![image](https://github.com/mkangoo/Telegram-bot-schedule/assets/121447049/102bbd06-f8c4-415b-81d6-22548f331742)
2.I came up with the idea of ​​doing something like `try().expect()` in jest.
I'll break these methods into two functions, just like you said.

Ilanaya commented on src/core/schedule.js:
```diff
@@ -67,3 +43,35 @@ export const getFullSchedule = shiftWeek => {
+	}
+}
+
+const holidayStatus = () => {
+	const dayIndex = getDayIndex().today()
+	return orderedWeekDays[dayIndex] === 'Sunday'
+}
```
1. If something is a boolean its name should start with a `is` or `has` prefix most of the time
2. Function name should start with a verb

I suggest `checkIsSunday`  for this function

mkangoo replied:
I'll fix it

mkangoo commented:
@Ilanaya I worked at night and did not push, I fixed some of the problems you described, I selected the correct names instead of {raw}. I need a little time, I will make the changes you suggest and ask for a review.

mkangoo commented:
@Ilanaya I've made the changes, but I still think there is room for improvement.

Ilanaya commented on src/core/schedule.js:
```diff
@@ -67,3 +42,31 @@ export const getFullSchedule = shiftWeek => {
+export const getTomorrowSchedule = () => {
+	const lastDayIndex = getTomorrowIndex()
+	const checkOnSunday = checkIsSunday(getTodayIndex(new Date()))
+	return getLessonsForDay(orderedWeekDays[lastDayIndex], dataBase, { weekOffset: checkOnSunday })
```
```suggestion
	const isSunday = checkIsSunday(getTodayIndex(new Date()))
	return getLessonsForDay(orderedWeekDays[lastDayIndex], dataBase,  isSunday ? getWeekNumber(new Date()) + 1 : getWeekNumber())
```

Ilanaya commented on src/core/schedule.js:
```diff
@@ -25,21 +16,12 @@ export const getScheduleOutput = (lessonType, { start, end, title, description,
-
-export function getLessonsForDay(targetDay, scheduleData, shouldShiftWeek = false, weekNumber = getWeekNumber(new Date())) {
+export function getLessonsForDay(targetDay, scheduleData, statusOffset, weekNumber = getWeekNumber(new Date())) {
 	const lessonToday = scheduleData.find(({ day, events }) => day === targetDay && events)
```
Why do we even need this `statusOffset` param? 
1. It's name is really confusing and `weekOffset` looks bad too.

As I can see it only adds `+ 1` to the current week number if user passes `{ weekOffset: true }`. This function already has `weekNumber` param to control from which week to take schedule from.

I think we can easily remove this param at all. Check comment above for an example

mkangoo commented:
1.I noticed an error in the operation of the `weekSchedule` and `nextWeekSchedule` commands. The problem was related to the flag being passed. I fixed it.
2.I made changes and removed the extra flag in the `getLessonsForDay` function.
I also changed the transmitted parameters.

mkangoo commented on src/commands/registerBotCommands.js:
```diff
@@ -34,7 +35,11 @@ export default async bot => {
-		{ command: 'next_week_schedule', handler: getWeekScheduleHandler({ weekOffset: true }), description: 'Расписание на неделю' },
+		{
+			command: 'next_week_schedule',
+			handler: getWeekScheduleHandler({ weekOffset: true }),
```
I can't think of a better name, since `week offset' is a direct translation of what I mean. Perhaps you can tell me how best to proceed?

Ilanaya replied:
Ok, you have `getWeekScheduleHandler` method. What would you expect it to do?

I suppose it should return some kind of schedule (which is already kinda working) and have some configuration. As we don't directly mention from which week it should get the schedule it would be reasonalble to expect it to take the **week number** as parameter.

You already have the method to determine the current week number by current day - so you can just remove this boolean params and pass `weekNumber` everywhere.

mkangoo replied:
Ok, I made changes to the code. Yes, you are right, this is much better

mkangoo commented:
Should we work on this pull request or can I merge it with the master?
